frontend/apps/reader/modules/readerthumbnail.lua

```diff
@@ -261,14 +261,15 @@ function ReaderThumbnail:cancelPageThumbnailRequests(batch_id)
     end
 end
 
-function ReaderThumbnail:getPageThumbnail(page, width, height, batch_id, when_generated_callback)
+function ReaderThumbnail:getPageThumbnail(page, page_hash, width, height, batch_id, when_generated_callback)
     self:setupCache()
     self.current_target_size_tag = string.format("w%d_h%d", width, height)
     if self.ui.rolling and Screen.night_mode and self.ui.document.configurable.nightmode_images == 1 then
         -- We'll get a different bb in this case: it needs its own cache hash
         self.current_target_size_tag = self.current_target_size_tag .. "_nm"
     end
-    local hash = string.format("p%d-%s", page, self.current_target_size_tag)
+    local hash = "thumb"..page_hash:gsub("|", "-").."-"..self.current_target_size_tag
+    -- local hash = "thumb"..page_hash.."|"..self.current_target_size_tag
```

Comment:
This is what [I'm talking about](https://github.com/koreader/koreader/pull/12085#issuecomment-2227469665): when keeping `|` in the hash, caching does not work: opening, closing, and then reopening the page browser result in the thumbnails being re-generated again.

Reply:
Possibly because of this?:
https://github.com/koreader/koreader/blob/7ebd067a95886392b3894bed0d4a0c50de475b59/frontend/apps/reader/modules/readerthumbnail.lua#L245-L250
and the versatile:
https://github.com/koreader/koreader/blob/7ebd067a95886392b3894bed0d4a0c50de475b59/frontend/apps/reader/modules/readerthumbnail.lua#L171-L201

Reply:
Keeping your | could be fine (and safer if we are sure there is no "-"), if you keep my `.."-"..self.current_target_size_tag`

Reply:
Except there's also:
```lua
    if start_page and end_page then
        local hash_subs_to_remove = {}
        for p=start_page, end_page do
            table.insert(hash_subs_to_remove, self.ui.dostring.format("p%d-", p))
        end
        self:removeFromCache(hash_subs_to_remove)
    end
```

Reply:
You could just call ReaderThumbnail:resetCache() on color change - it's not something a user will change often, so we may be fine with trashing it all.
Aren't we already doing it here ?!:
https://github.com/koreader/koreader/blob/7ebd067a95886392b3894bed0d4a0c50de475b59/frontend/apps/reader/modules/readerthumbnail.lua#L545-L548

Reply:
Oh great, another setting…